update data source permissions instructions to match modern UI (#63405)
grafana/grafana: docs/sources/administration/data-source-management/index.md

1. Navigate to **Configuration > Data Sources**.

> 1. Select the data source you want to assign query permissions for.

Reviewer: ```suggestion
1. Select the data source you want to assign query permissions to.
```

> 1. On the Permissions tab, click **Add Permission**.

Reviewer: ```suggestion
1. On the **Permissions** tab, click **Add Permission**.
```

> 1. Select **Team**, **User**, or **Role**

Reviewer: ```suggestion
1. Select **Team**, **User**, or **Role**.
```

> 1. Select the entity you want to modify permissions for

Reviewer: ```suggestion
1. Select the entity you want to modify permissions for.
```

> 2. Select the permission you would like to grant the entity, either **Query** or **Edit**

Reviewer: ```suggestion
2. Select the **Query** or **Edit** permission you want to grant the entity.
```

> 3. Click **Save**.

Reviewer: ```suggestion
1. Click **Save**.
```
